Mathlib/Topology/ProperMap.lean

```diff
@@ -226,6 +254,45 @@ theorem isProperMap_iff_isClosedMap_and_compact_fibers :
     rw [← clusterPt_lift'_closure_iff]
     exact hx.mono inf_le_left
 
+/-- An injective and continuous function is proper if and only if it is closed. -/
+lemma isProperMap_iff_isClosedMap_of_inj (f_cont : Continuous f) (f_inj : f.Injective) :
+    IsProperMap f ↔ IsClosedMap f := by
+  refine ⟨fun h ↦ h.isClosedMap, fun h ↦ ?_⟩
+  rw [isProperMap_iff_isClosedMap_and_compact_fibers]
+  exact ⟨f_cont, h, fun y ↦ (subsingleton_singleton.preimage f_inj).isCompact⟩
+
+/-- A injective continuous and closed map is proper. -/
+lemma isProperMap_of_isClosedMap_of_inj (f_cont : Continuous f) (f_inj : f.Injective)
+    (f_closed : IsClosedMap f) : IsProperMap f :=
+  (isProperMap_iff_isClosedMap_of_inj f_cont f_inj).2 f_closed
+
+/-- A homeomorphism is proper. -/
+@[simp] lemma Homeomorph.isProperMap (e : X ≃ₜ Y) : IsProperMap e :=
+  isProperMap_of_isClosedMap_of_inj e.continuous e.injective e.isClosedMap
+
+/-- The identity is proper. -/
+@[simp] lemma isProperMap_id : IsProperMap (id : X → X) := (Homeomorph.refl X).isProperMap
+
+/-- The coercion from a closed subset is proper. -/
+lemma isProperMap_subtype_val_of_closed {U : Set X} (hU : IsClosed U) : IsProperMap ((↑) : U → X) :=
+  isProperMap_of_isClosedMap_of_inj
+    continuous_subtype_val
+    Subtype.val_injective
+    hU.closedEmbedding_subtype_val.isClosedMap
```

Comment:
Last small comment from me: since you're using `hU.closedEmbedding_subtype_val`, why not use the `ClosedEmbedding` version of the lemma?